pkg/operator/ceph/object/user/controller.go

```diff
@@ -191,7 +191,22 @@ func (c *ObjectStoreUserController) createUser(context *clusterd.Context, u *cep
 
 	user, rgwerr, err := object.CreateUser(objContext, userConfig)
 	if err != nil {
-		return fmt.Errorf("failed to create user %s. RadosGW returned error %d: %+v", u.Name, rgwerr, err)
+		if rgwerr == object.RGWErrorBadData {
```

Comment:
What would cause the RGWErrorBadData error? Why wouldn't we retry in that case?

Reply:
Invalid/Existing username/email causes it and we do not want to retry because that can not be automatically fixed. It will require human intervention.

Reply:
Makes sense. To remove the duplicated error checking code, how about if we just remove the error check on lines 194-196? This means it will try once more before aborting, but this seems better than the duplication.

Reply:
Ack.